File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/WeatherRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.remote.WeatherResponse
import com.example.nutrismart.data.remote.WeatherRetrofit
import retrofit2.Response
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class WeatherRepository @Inject constructor() {
    suspend fun getCurrentWeather(lat: Double, lon: Double): Response<WeatherResponse> {
        return WeatherRetrofit.api.getCurrentWeather(lat, lon)
    }
}

File: NutriSmartIOS/data/repository/WeatherRepository.swift
//
//  WeatherRepository.swift
//  NutriSmartIOS
//

import Foundation

class WeatherRepository {
    static let shared = WeatherRepository()
    private let weatherService = WeatherService.shared
    
    private init() {}
    
    func getCurrentWeather(lat: Double, lon: Double) async throws -> WeatherResponse {
        return try await weatherService.getCurrentWeather(lat: lat, lon: lon)
    }
}
